framework/wazuh/core/rootcheck.py

```diff
@@ -0,0 +1,105 @@
+        if statuses:
+            for status in statuses:
+                self.query_filters.remove(status)
+            first_status = statuses[0]
+            first_status['separator'] = 'AND' if first_status['separator'] == '' else first_status['separator']
+            self.query_filters.insert(0, statuses[0])
+            self.query_filters[-1]['separator'] = ''
+
+    def _filter_status(self, filter_status):
+        partial = "SELECT {0} AS status, date_first, date_last, log, pci_dss, cis FROM pm_event AS t WHERE " \
+                  "date_last {1} (SELECT date_last-86400 FROM pm_event WHERE log = 'Ending rootcheck scan.')"
+        log_not_in = ") WHERE log NOT IN ('Starting rootcheck scan.', 'Ending rootcheck scan.', " \
+                     "'Starting syscheck scan.', 'Ending syscheck scan.'"
+
+        if filter_status['value'] == 'all':
+            self.query += partial.format("'outstanding'", '>') + " UNION " + partial.format("'solved'", '<=') + log_not_in
+        elif filter_status['value'] == 'outstanding':
+            self.query += partial.format("'outstanding'", '>') + log_not_in
+        elif filter_status['value'] == 'solved':
+            self.query += partial.format("'solved'", '<=') + log_not_in
+        else:
+            raise WazuhException(1603, filter_status['value'])
+
+    def _format_data_into_dictionary(self):
+        def format_fields(field_name, value):
+            if field_name in ['date_first', 'date_last']:
+                return datetime.utcfromtimestamp(value).strftime("%Y-%m-%d %H:%M:%S")
+            else:
+                return value
+
+        return {'items': [{field: format_fields(field, db_tuple[field]) for field in self.select |
+                           self.min_select_fields if field in db_tuple and db_tuple[field] is not None}
+                          for db_tuple in self._data], 'totalItems': self.total_items}
+
+    @staticmethod
+    def _pass_filter(db_filter):
+        return False
+
+
+def last_scan(agent_id):
+    """Gets the last rootcheck scan of an agent.
+
+    :param agent_id: Agent ID.
+    :return: Dictionary: end, start.
+    """
+    Agent(agent_id).get_basic_information()
+    wdb_conn = WazuhDBConnection()
+    data = {}
+
+    # end time
+    result = wdb_conn.execute(f"agent {agent_id} sql SELECT max(date_last) FROM pm_event WHERE "
+                              "log = 'Ending rootcheck scan.'")
+    time = list(result[0].values())[0] if result else None
+    data['end'] = datetime.utcfromtimestamp(time).strftime("%Y-%m-%d %H:%M:%S") if time is not None else None
+
+    # start time
+    result = wdb_conn.execute(f"agent {agent_id} sql SELECT max(date_last) FROM pm_event "
+                              "WHERE log = 'Starting rootcheck scan.'")
+    time = list(result[0].values())[0] if result else None
+    data['start'] = datetime.utcfromtimestamp(time).strftime("%Y-%m-%d %H:%M:%S") if time is not None else None
+
+    return data if data else None
```

Comment:
What last scan information do we return if a scan has already started but not finished? End time can never be older than Start time, check syscheck SDK.